aviary/mission/problem_builder_height_energy.py

```diff
@@ -0,0 +1,443 @@
+        )
+
+        # Build and add takeoff subsystem
+        takeoff = takeoff_options.build_phase(False)
+        prob.model.add_subsystem(
+            'takeoff',
+            takeoff,
+            promotes_inputs=['aircraft:*', 'mission:*'],
+            promotes_outputs=['mission:*'],
+        )
+
+    def get_phase_builder(self, prob, phase_name, phase_options):
+
+        if 'phase_builder' in phase_options:
+            phase_builder = phase_options['phase_builder']
+            if not issubclass(phase_builder, PhaseBuilderBase):
+                raise TypeError(f"phase_builder for the phase called "
+                                "{phase_name} must be a PhaseBuilderBase object.")
+        else:
+            phase_builder = EnergyPhase
+
+        return phase_builder
+
+    def set_phase_options(self, prob, phase_name, phase_idx, phase, user_options):
+
+        try:
+            fix_initial = user_options.get_val('fix_initial')
+        except KeyError:
+            fix_initial = False
+
+        try:
+            fix_duration = user_options.get_val('fix_duration')
+        except KeyError:
+            fix_duration = False
+
+        # The rest of the phases includes all Height Energy method phases
+        # and any 2DOF phases that don't fall into the naming patterns
+        # above.
+        input_initial = False
+        time_units = phase.time_options['units']
+
+        # Make a good guess for a reasonable intitial time scaler.
+        try:
+            initial_bounds = user_options.get_val('initial_bounds', units=time_units)
+        except KeyError:
+            initial_bounds = (None, None)
+
+        if initial_bounds[0] is not None and initial_bounds[1] != 0.0:
+            # Upper bound is good for a ref.
+            user_options.set_val('initial_ref', initial_bounds[1],
+                                 units=time_units)
+        else:
+            user_options.set_val('initial_ref', 600., time_units)
+
+        duration_bounds = user_options.get_val("duration_bounds", time_units)
+        user_options.set_val(
+            'duration_ref', (duration_bounds[0] + duration_bounds[1]) / 2.,
+            time_units
+        )
+        if phase_idx > 0:
+            input_initial = True
+
+        if fix_initial or input_initial:
+
+            if prob.comm.size > 1:
+                # Phases are disconnected to run in parallel, so initial ref is
+                # valid.
+                initial_ref = user_options.get_val("initial_ref", time_units)
+            else:
+                # Redundant on a fixed input; raises a warning if specified.
+                initial_ref = None
+
+            phase.set_time_options(
+                fix_initial=fix_initial, fix_duration=fix_duration, units=time_units,
+                duration_bounds=user_options.get_val("duration_bounds", time_units),
+                duration_ref=user_options.get_val("duration_ref", time_units),
+                initial_ref=initial_ref,
```

Comment:
why are we not applying the initial_bounds that the user supplied here??

Reply:
It is potentially an oversight. We should treat initial_bounds just like initial_ref -- none unless mpi.